lib/src/ruby/ruby.cc

```diff
@@ -18,14 +18,14 @@ static const char load_puppet[] =
 "  $LOAD_PATH << Puppet[:libdir]\n"
 "end\n"
 "Facter.reset\n"
+"Facter.search_external([Puppet[:pluginfactdest]])\n"
```

Comment:
Just to confirm, puppet 3.x includes facts from 5 sources:

* non-pluginsynced external facts (`/etc/facter/facts.d`)
* module external (`/etc/puppet/modules/foo/facts.d/`) and custom (`/etc/puppet/modules/foo/lib/facter`) facts
* pluginsynced external (`/var/lib/puppet/facts.d/`) and custom (`/var/lib/puppet/lib/facter`) facts

facter 2.x only reports on

* non-pluginsynced external facts (`/etc/facter/facts.d`)
* custom (`/etc/puppet/modules/foo/lib/facter`) facts

```
$ sudo bundle exec puppet facts find . | jq .values | grep myfact | sort
  "myfact_external": "true",
  "myfact_module_custom": "true",
  "myfact_module_external": "true",
  "myfact_pluginsync_custom": "true",
  "myfact_pluginsync_external": "true",
$ sudo bundle exec facter -p | grep myfact | sort
myfact_external => true
myfact_pluginsync_custom => true
```

Also both puppet 3.x and facter 2.x include `puppetversion`:

```
$ sudo bundle exec facter -p | grep puppetversion
puppetversion => 3.8.4
$ sudo bundle exec puppet facts find . | jq .values | grep puppetversion
  "puppetversion": "3.8.4",
```

We fixed facter (in 3.x series) to include pluginsynced external facts, so both puppet and facter return the same set of fact excluding custom and external facts from modules:

```
# /opt/puppetlabs/bin/puppet facts find  | grep myfact
    "myfact_external": "true",
    "myfact_module_custom": true,
    "myfact_module_external": "true",
    "myfact_pluginsync_custom": true,
    "myfact_pluginsync_external": "true",
# /opt/puppetlabs/bin/facter -p  | grep myfact
myfact_external => true
myfact_pluginsync_custom => true
myfact_pluginsync_external => true
```

And without this PR, `facter -p` omits puppetversion:

```
# /opt/puppetlabs/bin/facter -p | grep puppetversion
# /opt/puppetlabs/bin/puppet facts find  | grep puppetversion
    "puppetversion": "4.3.0",
```

So :+1: